### PR TITLE
Add support for ssh multiplexing

### PR DESCRIPTION
**Pull request by RHEmployee**
Add support for ssh multiplexing.

### CODE REVIEW

**Comment by lukaszachy** [comment]
Seems that this PR works well - I've tried connect and virtual (both system and session)

File: tmt/steps/provision/__init__.py [hunk 1]
```diff
@@ -251,19 +255,41 @@ def _ssh_options(self, join=False):
                 options.extend(['-i', shlex.quote(key) if join else key])
         if self.password:
             options.extend(['-oPasswordAuthentication=yes'])
+
+        if not self._ssh_socket_path:
+            self._ssh_socket_path = f"/run/user/{os.getuid()}/tmt/" + next(
+                tempfile._get_candidate_names())
```
[the remark this answers is not in the document]
Comment (psss) [reply tmt/steps/provision/__init__.py:264]: Yes, the traceback is gone.

**Review by psss — APPROVED** [review]
I've done a couple of adjustments in 741e843 which should make the code flow a bit more simple. Please, have a look and check if I haven't broken anything essential.

File: tmt/steps/provision/__init__.py [hunk 2]
```diff
@@ -236,6 +242,14 @@ def _ssh_guest(self):
         """ Return user@guest """
         return f'{self.user}@{self.guest}'
 
+    def _ssh_socket(self):
+        """ Prepare path to the master connection socket """
+        if not self._ssh_socket_path:
+            self._ssh_socket_path = os.path.join(
```
Comment (lukaszachy) [inline tmt/steps/provision/__init__.py:248]: How about following instead?  
```python
    def _ssh_socket(self):
        """ Prepare path to the master connection socket """
        if not self._ssh_socket_path:
            socket_dir = f"/run/user/{os.getuid()}/tmt"
            os.makedirs(socket_dir, exist_ok=True)
            self._ssh_socket_path = tempfile.mktemp(dir=socket_dir)
        return self._ssh_socket_path
```

Comment (psss) [reply tmt/steps/provision/__init__.py:248]: Sounds good! Updated.

**Comment by psss** [comment]
Maybe in the future we could take some more inspiration from cockpits' [SSHConnection](https://github.com/cockpit-project/bots/blob/main/machine/machine_core/ssh_connection.py) class as @jscotka suggested. Or even re-use it? But for now, as the first step, I think it's good enough. Tests are green.

**Review by lukaszachy — APPROVED** [review]

**Comment by RHEmployee** [comment]
Hi. Thank you very much! Great inspirative changes.

**Pull request merged**